avm/res/managed-services/registration-definition/main.bicep

```diff
@@ -0,0 +1,96 @@
+metadata name = 'Registration Definitions'
+metadata description = '''This module deploys a `Registration Definition` and a `Registration Assignment` (often referred to as 'Lighthouse' or 'resource delegation')
+on subscription or resource group scopes. This type of delegation is very similar to role assignments but here the principal that is
+assigned a role is in a remote/managing Azure Active Directory tenant. The templates are run towards the tenant where
+the Azure resources you want to delegate access to are, providing 'authorizations' (aka. access delegation) to principals in a
+remote/managing tenant.'''
+metadata owner = 'Azure/module-maintainers'
+
+targetScope = 'subscription'
+
+@description('Required. Specify a unique name for your offer/registration. i.e \'<Managing Tenant> - <Remote Tenant> - <ResourceName>\'.')
+param name string
+
+@description('Required. Description of the offer/registration. i.e. \'Managed by <Managing Org Name>\'.')
+param registrationDescription string
+
+@description('Required. Specify the tenant ID of the tenant which homes the principals you are delegating permissions to.')
+param managedByTenantId string
+
+@description('Required. Specify an array of objects, containing object of Azure Active Directory principalId, a Azure roleDefinitionId, and an optional principalIdDisplayName. The roleDefinition specified is granted to the principalId in the provider\'s Active Directory and the principalIdDisplayName is visible to customers.')
+param authorizations array
+
+@description('Optional. Specify the name of the Resource Group to delegate access to. If not provided, delegation will be done on the targeted subscription.')
+param resourceGroupName string = ''
+
+@description('Optional. Location deployment metadata.')
+param resourceLocation string = deployment().location
```

Comment:
Would not recommend to call it `resourceLocation` if it's not a resourceLocation but either something like 
- location
- metadataLocation
- telemetryLocation

Reply:
Wait, so it's used everywhere as `resourceLocation` but here I shouldn't?

Reply:
Hey @segraef it 'should' be used everywhere as a resource location. Like the deployment of a key vault which is deployed into a location. In this case it's only used for the telemetry deployment object, or isn't it?

Reply:
@AlexanderSehr, changed.